File: app/src/main/java/amagi82/modularcharactersheetcreator/ui/sheet/modules/HealthViewModel.java
package amagi82.modularcharactersheetcreator.ui.sheet.modules;

import android.databinding.ObservableField;

import amagi82.modularcharactersheetcreator.models.modules.Health;
import amagi82.modularcharactersheetcreator.models.modules.Module;
import amagi82.modularcharactersheetcreator.ui._base.BaseModuleViewModel;

public class HealthViewModel extends BaseModuleViewModel {
    public final ObservableField<Health> health = new ObservableField<>();

    public HealthViewModel(Module module) {
        super(module);
        update(module);
    }

    public void update(Module module){
        health.set(module.health());
    }
}

File: app/src/main/java/amagi82/modularcharactersheetcreator/ui/sheet/modules/HealthViewModel.kt
package amagi82.modularcharactersheetcreator.ui.sheet.modules

import amagi82.modularcharactersheetcreator.models.modules.Health
import amagi82.modularcharactersheetcreator.models.modules.Module
import amagi82.modularcharactersheetcreator.ui._base.BaseModuleViewModel
import android.databinding.ObservableField

class HealthViewModel(module: Module) : BaseModuleViewModel(module) {
    val health = ObservableField<Health>()

    init {
        update(module)
    }

    override fun update(module: Module) {
        health.set(module.health)
    }
}
